lib/core/src/restore.rs

```diff
@@ -0,0 +1,1164 @@
+    /// and in some cases, between [PaymentState::Created] and [PaymentState::Failed].
+    fn get_partial_state(&self) -> PaymentState;
+}
+
+pub(crate) struct RecoveredOnchainDataSend {
+    lockup_tx_id: Option<HistoryTxId>,
+    claim_tx_id: Option<HistoryTxId>,
+    refund_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataSend {
+    fn get_partial_state(&self) -> PaymentState {
+        match &self.lockup_tx_id {
+            Some(_) => match &self.claim_tx_id {
+                Some(_) => PaymentState::Complete,
+                None => match &self.refund_tx_id {
+                    Some(refund_tx_id) => match refund_tx_id.confirmed {
+                        true => PaymentState::Failed,
+                        false => PaymentState::RefundPending,
+                    },
+                    None => PaymentState::Pending,
+                },
+            },
+            None => PaymentState::TimedOut,
+        }
+    }
+}
+
+pub(crate) struct RecoveredOnchainDataReceive {
+    lockup_tx_id: Option<HistoryTxId>,
+    claim_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataReceive {
+    fn get_partial_state(&self) -> PaymentState {
+        match (&self.lockup_tx_id, &self.claim_tx_id) {
+            (Some(_), Some(claim_tx_id)) => match claim_tx_id.confirmed {
+                true => PaymentState::Complete,
+                false => PaymentState::Pending,
+            },
+            (Some(_), None) => PaymentState::Pending,
+            // TODO How to distinguish between Failed and Created (if in both cases, no lockup or claim tx present)
+            //   See https://docs.boltz.exchange/v/api/lifecycle#reverse-submarine-swaps
+            _ => PaymentState::Failed,
+        }
+    }
+}
+
+pub(crate) struct RecoveredOnchainDataChainSend {
+    lbtc_user_lockup_tx_id: Option<HistoryTxId>,
+    lbtc_refund_tx_id: Option<HistoryTxId>,
+    btc_server_lockup_tx_id: Option<HistoryTxId>,
+    btc_claim_tx_id: Option<HistoryTxId>,
+}
+impl PartialSwapState for RecoveredOnchainDataChainSend {
+    fn get_partial_state(&self) -> PaymentState {
+        // TODO How to detect TimedOut state?
+        //     TimedOut: This covers the case when the swap state is still Created and the swap fails to reach the
+        //     Pending state in time. The TimedOut state indicates the lockup tx should never be broadcast.
+        match &self.lbtc_user_lockup_tx_id {
+            Some(_) => match &self.btc_claim_tx_id {
+                Some(_) => PaymentState::Complete,
+                None => match &self.lbtc_refund_tx_id {
+                    Some(tx) => match tx.confirmed {
+                        true => PaymentState::Failed,
+                        false => PaymentState::RefundPending,
+                    },
+                    None => PaymentState::Created,
+                },
+            },
+            None => PaymentState::Created,
```

Comment:
Same as with send swap this should always be Timeout as on send we automatically attempt to pay and never leave the swap in Created state.
We also ensure the Created ones are converted to Timeout on startup: https://github.com/breez/breez-sdk-liquid/blob/main/lib/core/src/sdk.rs#L184

Reply:
Done: 07759523cb6f57ecee90eaf72a4d9dee591a21ab